lib/money/parser/simple.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+class Money
+  module Parser
+    class Simple
+      class << self
+        # Parses an input string using BigDecimal, it always expects a dot character as a decimal separator and
+        # generally does not accept other characters other than minus-hyphen and digits. It is useful for APIs, interop
+        # with other languages and other use cases where you expect well-formatted input and do not need to take user
+        # locale into consideration.
+        # @param input [String]
+        # @param currency [String, Money::Currency]
+        # @param strict [Boolean]
+        # @return [Money, nil]
+        def parse(input, currency, strict: false)
+          currency = Money::Helpers.value_to_currency(currency)
+          return unless currency
+
+          amount = BigDecimal(input, exception: false)
```

Comment:
Calling Money directly adds the complexity of an app potentially having opted into deprecated behaviour and returning `0` instead of `nil` for invalid input, this doesn't work with the expectations from our parameter type coercion gem that is used in Shopify core and other apps.

This parser is meant for well-formatted data so I am wary of normalizing here to paper over bugs from other people. We can loosen restrictions later if we learn it's needed (the opposite is much harder).

Good call on the scientific notation, I added something for that 😄

Reply:
> opted into deprecated behaviour and returning 0 instead of nil

If we're already expecting a well formatted string, could we get them to not opt-in to that deprecated behaviour, instead of adding a new parser? IMO nil -> 0 should already never be happening and is fairly easy to do manually if needed `Money.new(amount || 0)`

Reply:
I am not sure I follow. Are you suggesting to remove the deprecation for invalid input for Money.new? 

Assuming an app hasn't opted in to the deprecated path, Money.new accepts more input than we want for APIs. Money.new will raise on invalid input where our typed parameter code (see related PR) and GraphQL types need a `nil` returned. We could rescue for those cases but that's less performant for most inputs.

I don't understand how `Money.new(amount || 0)` is of use, but if you meant `Money::Parser::Simple.parse(amount) || 0` then yes that's another reason to prefer returning `nil`.